templates/step_function/backend_checks_definition.json.tpl

```diff
@@ -0,0 +1,279 @@
+              "antivirus.$": "$.[?(@.antivirus)].antivirus",
+              "fileFormat.$": "$.[?(@.fileFormat)].fileFormat",
+              "checksum.$": "$.[?(@.checksum)].checksum"
+            }
+          }
+        }
+      },
+      "MaxConcurrency": 40,
+      "ResultSelector": {
+        "results.$": "$"
+      },
+      "ItemsPath": "$.results",
+      "Next": "Redacted files",
+      "Catch": [
+        {
+          "ErrorEquals": [
+            "States.ALL"
+          ],
+          "Next": "Notify"
+        }
+      ]
+    },
+    "Redacted files": {
+      "Type": "Task",
+      "Resource": "arn:aws:states:::lambda:invoke",
+      "Parameters": {
+        "Payload.$": "$",
+        "FunctionName": "${redacted_files_arn}"
+      },
+      "Retry": [
+        {
+          "ErrorEquals": [
+            "Lambda.ServiceException",
+            "Lambda.AWSLambdaException",
+            "Lambda.SdkClientException",
+            "Lambda.TooManyRequestsException"
+          ],
+          "IntervalSeconds": 2,
+          "MaxAttempts": 6,
+          "BackoffRate": 2
+        }
+      ],
+      "Next": "Generate Statuses",
+      "ResultPath": "$.redactedResults",
+      "ResultSelector": {
+        "redactedFiles.$": "$.Payload.redactedFiles",
+        "errors.$": "$.Payload.errors"
+      }
+    },
+    "Generate Statuses": {
+      "Type": "Task",
+      "Resource": "arn:aws:states:::lambda:invoke",
+      "Parameters": {
+        "Payload.$": "$",
+        "FunctionName": "${statuses_arn}"
+      },
+      "Retry": [
+        {
+          "ErrorEquals": [
+            "Lambda.ServiceException",
+            "Lambda.AWSLambdaException",
+            "Lambda.SdkClientException",
+            "Lambda.TooManyRequestsException"
+          ],
+          "IntervalSeconds": 2,
+          "MaxAttempts": 6,
+          "BackoffRate": 2
+        }
+      ],
+      "Next": "Update API",
+      "ResultPath": "$.statuses",
+      "ResultSelector": {
+        "statuses.$": "$.Payload.statuses"
+      }
+    },
+    "Update API": {
+      "Type": "Task",
+      "Resource": "arn:aws:states:::lambda:invoke",
+      "OutputPath": "$.Payload",
+      "Parameters": {
+        "Payload.$": "$",
+        "FunctionName": "${api_update_v2_arn}"
+      },
+      "Retry": [
+        {
+          "ErrorEquals": [
+            "Lambda.ServiceException",
+            "Lambda.AWSLambdaException",
+            "Lambda.SdkClientException"
+          ],
+          "IntervalSeconds": 2,
+          "MaxAttempts": 6,
+          "BackoffRate": 2
+        }
+      ],
+      "End": true
+    },
+    "Notify": {
+      "Type": "Task",
+      "Resource": "arn:aws:states:::lambda:invoke",
+      "OutputPath": "$.Payload",
+      "Parameters": {
+        "FunctionName": "${notification_arn}",
+        "Payload": {
+          "consignmentId.$": "$$.Execution.Input.consignmentId",
+          "error.$": "$.Error",
+          "cause.$": "$.Cause",
+          "environment": "intg"
```

Comment:
Should the `environment` field have a hard coded value?

Reply:
Good spot. I have updated it, thanks.